homeassistant/components/deconz/config_flow.py

```diff
@@ -161,7 +161,11 @@ def __init__(self):
 
     def _update_entry(self, entry, host, port, api_key=None):
         """Update existing entry."""
-        if entry.data[CONF_HOST] == host:
+        if (
+            entry.data[CONF_HOST] == host
+            and entry.data[CONF_PORT] == port
+            and (api_key is None or entry.data[CONF_API_KEY] == api_key)
```

Comment:
When would updating the api key happen?

Reply:
Re-install the add-on? Reset the gateway?